File: app/src/main/java/com/loskon/noteminimalism3/auxiliary/bp/InternetCheck.java
package com.loskon.noteminimalism3.auxiliary.bp;

import android.app.Activity;
import android.content.Context;
import android.net.ConnectivityManager;
import android.os.Build;

/**
 * Проверка наличия интернет-соединения
 */

public class InternetCheck  {

    private final Activity activity;

    public InternetCheck(Activity activity) {
        this.activity = activity;
    }

    public boolean isConnected() {
        boolean status = false;

        ConnectivityManager connectivityManager =
                (ConnectivityManager) activity.getSystemService(Context.CONNECTIVITY_SERVICE);

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            if (connectivityManager != null && connectivityManager
                    .getActiveNetwork() != null && connectivityManager
                    .getNetworkCapabilities(connectivityManager.getActiveNetwork()) != null) {
                status = true;
            }
        } else {
            if (connectivityManager != null &&
                    connectivityManager.getActiveNetworkInfo() != null &&
                    connectivityManager.getActiveNetworkInfo().isConnectedOrConnecting()) {
                status = true;
            }
        }

        return status;
    }
}

File: app/src/main/java/com/loskon/noteminimalism3/other/InternetCheck.kt
package com.loskon.noteminimalism3.other

import android.content.Context
import android.net.ConnectivityManager
import android.net.Network
import android.net.NetworkCapabilities
import android.os.Build
import androidx.annotation.RequiresApi

/**
 * Проверка интернет-соединения
 */

object InternetCheck {

    fun isConnected(context: Context): Boolean {
        val connectivityManager: ConnectivityManager = context
            .getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager

        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            checkConnectedAndroidM(connectivityManager)
        } else {
            checkConnected(connectivityManager)
        }
    }

    @RequiresApi(Build.VERSION_CODES.M)
    private fun checkConnectedAndroidM(connectivityManager: ConnectivityManager): Boolean {
        val network: Network = connectivityManager
            .activeNetwork ?: return false

        val activeNetwork: NetworkCapabilities = connectivityManager
            .getNetworkCapabilities(network) ?: return false

        return when {
            activeNetwork.hasTransport(NetworkCapabilities.TRANSPORT_WIFI) -> true
            activeNetwork.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR) -> true
            else -> false
        }
    }

    @Suppress("DEPRECATION")
    private fun checkConnected(connectivityManager: ConnectivityManager): Boolean {
        val networkInfo = connectivityManager.activeNetworkInfo ?: return false
        return networkInfo.isConnected
    }
}

